iabgpp-encoder/src/main/java/com/iab/gpp/encoder/segment/UsCoV1CoreSegment.java

```diff
@@ -128,12 +128,12 @@ public void validate() {
             + targetedAdvertisingOptOutNotice + " / " + targetedAdvertisingOptOut + "}");
       }
     } else if (targetedAdvertisingOptOutNotice == 1) {
-      if (saleOptOut != 1 && saleOptOut != 2) {
+      if (targetedAdvertisingOptOut != 1 && targetedAdvertisingOptOut != 2) {
         throw new ValidationException("Invalid usco targeted advertising notice / opt out combination: {"
             + targetedAdvertisingOptOutNotice + " / " + targetedAdvertisingOptOut + "}");
       }
     } else if (targetedAdvertisingOptOutNotice == 2) {
-      if (saleOptOut != 1) {
+      if (targetedAdvertisingOptOut != 1) {
         throw new ValidationException("Invalid usco targeted advertising notice / opt out combination: {"
             + targetedAdvertisingOptOutNotice + " / " + targetedAdvertisingOptOut + "}");
       }
```

Comment:
This looks to be a logical error using the wrong optOut variable. Repeated below in other Us*V1CoreSegment#validate methods.